PetAdoptions/petsite/petsite/Controllers/PaymentController.cs

```diff
@@ -169,0 +1,104 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Amazon.SQS;
+using Microsoft.Extensions.Configuration;
+using Prometheus;
+
+namespace PetSite.Controllers
+{
+    public class PaymentController : BaseController
+    {
+        private static string _txStatus = String.Empty;
+        
+        private readonly ILogger<PaymentController> _logger;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IConfiguration _configuration;
+
+        //Prometheus metric to count the number of Pets adopted
+        private static readonly Counter PetAdoptionCount =
+            Metrics.CreateCounter("petsite_petadoptions_total", "Count the number of Pets adopted");
+
+        public PaymentController(ILogger<PaymentController> logger, IConfiguration configuration, IHttpClientFactory httpClientFactory)
+        {
+            _configuration = configuration;
+            _httpClientFactory = httpClientFactory;
+            _logger = logger;
+        }
+
+        // GET: Payment
+        [HttpGet]
+        public ActionResult Index()
+        {
+            if (EnsureUserId()) return new EmptyResult();
+            
+            // Transfer TempData to ViewData for the view
+            if (TempData["txStatus"] != null)
+            {
+                ViewData["txStatus"] = TempData["txStatus"];
+                ViewData["error"] = TempData["error"];
+            }
+            
+            return View();
+        }
+
+        // POST: Payment/MakePayment
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

MakePayment is a state-changing MVC method that does not validate the antiforgery token or do strict content-type checking. State-changing controller methods should either enforce antiforgery tokens or do strict content-type checking to prevent simple HTTP request types from bypassing CORS preflight controls. Learn more - https://cwe.mitre.org/data/definitions/352.html